### DISCUSSION TITLE
Kann man die LED deaktivieren?

### DISCUSSION THREAD

**Discussion by ZeroQool007** [frickelzeugs/FingerprintDoorbell#17] [Q&A]
Mega geiles Projekt! Hut ab.

Kann man die LED deaktivieren?

Gruß

**Comment by frickelzeugs** [comment]
Danke! Du meinst den runden Leuchtring? Der wird ja für Benutzerfeedback/Statusanzeige genutzt, daher ist das nicht vorgesehen den abzuschalten. Oder was hast du vor?

**Comment by jhubig** [comment]
Servus. Auch von mir vielen Dank für das tolle Projekt. Ist bereits nachgebaut. Liegt aber noch auf dem Schreibtisch.
Mich würde es auch interessieren ob man die LED deaktivieren kann. Für meinen Reader hatte ich es geplant, ihn vorne an der Garage anzubringen. Jetzt möchte ich nicht, dass das Teil zum Beispiel die ganze Nacht dort leuchtet. Wäre super wenn man das auch per MQTT Variable deaktivieren kann. Zum Beispiel in der Nacht. Oder auch tagsüber und dann per script aktivieren sobald ein Finger aufgelegt wird. Mir geht es hauptsächlich um die blaue LED solange sich der Reader im "System ready" Status befindet.

Ich denke prinzipiell wäre es möglich es komplett zu deaktivieren: https://esphome.io/components/fingerprint_grow.html#:~:text=Configuration%20options%3A-,state,-(Required%2C%20string

Super wäre es per MQTT auf off zu setzen und im Standardbetrieb auf AUTO (damit das breathing, flashing, etc. wieder normal funktioniert). Was denkst du?

Danke.

Grüße
Johannes

**Reply by jhubig** [reply]
Habe dein Repository jetzt mal "geforked" und etwas getestet. Habe ein neues MQTT Topic angelegt (LedTouchRing). Und natürlich auch die passenden Ergänzungen im Code.

In der FingerprintManager.cpp wird das nun in der setLedRingReady Methode geprüft:

![image](https://user-images.githubusercontent.com/7255837/200955914-af996fd6-005d-47a5-b8a4-a0582ebc006d.png)

Das lässt sich natürlich beliebig erweitern auf das Setzen der anderen states wie setLedRingError, setLedRingWifiConfig,... (überall wo LEDControl gesetzt wird)

Aber mir reicht das eigentlich so. Jetzt kann ich per MQTT per off oder on die LED einschalten (bisher eben nur für den ReadyState). Sie geht zeigt natürlich bisher weiterhin die anderen States per LED an.
Kann es aber gerne bei den anderen noch ergänzen und dann per Pull Request "einreichen", sofern das gewünscht ist

**Reply by jhubig** [reply]
Sorry. Es gibt ja schon einen passenden Pull-Request dafür. Etwas anders gelöst aber vergleichbares Ergebnis + zusätzliche Farben: https://github.com/frickelzeugs/FingerprintDoorbell/pull/26

**Reply by jhubig** [reply]
Habe das Setzen der Farbe sowie das Ausschalten der LED im Standby-Betrieb in meinem Fork implementiert falls jemand Interesse hat: https://github.com/jhubig/FingerprintDoorbell/commit/c9b2113dc6cc64b86cf10ccdd2cac71163e5652b

**Reply by baetzmr** [reply]
Super! Wird voraussichtlich so eingebaut. Warte nur noch auf den Sensor, damit ich einen Testaufbau machen kann.

**Reply by jhubig** [reply]
> Super! Wird voraussichtlich so eingebaut. Warte nur noch auf den Sensor, damit ich einen Testaufbau machen kann. 

Na dann viel Spaß 👍🏻😃
![image](https://user-images.githubusercontent.com/7255837/201953552-a1f8390d-7650-4623-a79d-9a4836a348cd.jpeg)

**Reply by baetzmr** [reply]
Kannst Du mal bitte prüfen ob bei Dir die mqtt Funktion ignoreTouchRing noch geht? Hab das bei mir mal so eingetragen und das ist das Einzige was nicht mehr funktioniert

**Reply by jhubig** [reply]
Hast du den letzten Stand aus meinem Commit genommen? Da heißt der neue Paramter LedTouchRingActive. Damit kannst du ihn eigentlich ausschalten.

**Reply by jhubig** [reply]
![image](https://user-images.githubusercontent.com/7255837/202261948-4646ecb4-58c1-4d81-a792-3800dedb5837.jpeg)

**Reply by baetzmr** [reply]
Das Ausschalten der LED geht super. Aber das ignoreTouchRing nicht mehr.

**Reply by jhubig** [reply]
Ah okay. Jetzt verstehe ich 😃.
Aber das hab ich bei den Änderungen nicht angefasst.

Wenn ich bei mir den
- ignoreTouchRing auf on setze dann kann ich den äußeren Ring berühren ohne dass er anfängt rot zu blinken
- ignoreTouchRing auf off setze dann kann ich den äußeren Ring berühren und er fängt an rot zu blinken

**Reply by baetzmr** [reply]
Ich schau es mir noch mal an. War vorhin etwas in Eile. Hab bestimmt was übersehen

**Reply by jhubig** [reply]
Jetzt erfolgreich gewesen?

**Reply by baetzmr** [reply]
Ja das mit dem ignoreTouch geht jetzt auch. Mir ist nur noch eine Sache aufgefallen. Wenn der LED Ring aus ist und Mann dann den ignoreTouchRing dazu nimmt, geht die LED wieder an. Ich schau mir das am Wochenende mal an.

**Reply by jhubig** [reply]
Ok. Super.

Ja. Ist noch nicht alles final getestet. Danke.

**Reply by baetzmr** [reply]
Kleine Änderung wenn ignoreTouchRing + LED abgeschaltet. Ich habe mal die fehlende Bedingung hizugefügt. Das funktioniert schon mal. Wie schon geschrieben habe ich ja noch das Problem das sporadisch die Klingel auslöst, obwohl ignoreTouchRing auf on steht.

![image](https://user-images.githubusercontent.com/81546284/202913980-f4952d74-d1b6-482c-adeb-5256fdb5176b.png)

**Reply by jhubig** [reply]
Stimmt. Die Bedingung hatte ich vergessen.
Ich benutze mein Setup nicht zum klingeln, daher kann ich da vermutlich nicht weiterhelfen. War das die Tage als es geregnet hat?

**Reply by baetzmr** [reply]
Der Sensor ist noch nicht eingebaut. Ist quasi noch in der Testphase bis ich alles in den Schaltschrank baue. Regen kann also ausgeschlossen werden. Ich denke mal es hat einen Einfluss wie lange der Finger aufgelegt wird. Dann wird manchmal ein noMatch Ereignis ausgelöst und mein Relais schaltet
![image](https://user-images.githubusercontent.com/81546284/202919648-aefa2ade-901d-4c25-9db0-376c130a3201.jpeg)

**Reply by jhubig** [reply]
Das sieht super aus 👍🏻.
Ja das kann sein. Ist mir auch schonmal im Log aufgefallen.

**Reply by baetzmr** [reply]
So jetzt geht alles. Ich habe für den ignoreTouchRing einen kleinen Workaround gemacht und eine zusätzliche Variable eingefügt. Außerdem habe ich noch die Möglichkeit hinzugefügt per mqtt die LED Sequenz schalten zu können (dauer An / Breathing / usw). Bis jetzt sind alle Tests in jeder Situation in Ordnung.

Danke an alle für dieses schöne Projekt!

**Reply by jhubig** [reply]
Schön zu hören. Hast du den Code "nur" bei dir lokal? Mich würde interessieren was du bei dir eingebaut hast.

Auch ein großer Dank von mir. Endlich mal wieder ein tolles Projekt (entdeckt bei Spiel und Zeug bei YouTube: https://youtu.be/MCayBntBlqk)

**Reply by baetzmr** [reply]
Ist bei mir lokal. Ich lad später wenn ich wieder daheim bin mal meine Änderungen hoch

**Reply by RolloD85** [reply]
Hallo baetzmr,

ich bin gerade über deinen Code gestolpert. Kannst du eventuell nochmal auflisten was du alles eingefügt bzw. geändert hast und gegeben falls dann die firmware.bin und die spiffs.bin zur Verfügung stellen.
Vielen Dank für deine Arbeit.

Gruß RolloD85
